### PR TITLE
Setup overhaul

### PR DESCRIPTION
**Pull request by GiovanH**
Page-based setup screen, with up-to-date settings and spoiler controls.

### CODE REVIEW

File: src/components/SystemPages/NewReaderControls.vue [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    },
+    handleEnable: {},
+    handleDisable: {},
+    handleChange: {}
+  },
+  data: function() {
+    return {
+      // The number in the input field. May be an mspa number or viz number depending on settings. Mutable.
+      // TODO: This assignment sometimes doesn't work, and assigns 001901 anyway?
+      newReaderPageInput: this.$newReaderCurrent || this.$mspaOrVizNumber('001901'),
```
Comment (Bambosh) [inline src/components/SystemPages/NewReaderControls.vue:150]: $newReaderCurrent defaults to 001901 out of a factory reset, then gets parsed as a viz number. Not ideal!

File: src/components/SystemPages/NewReaderControls.vue [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+          <p>New reader mode disabled.</p>
+          <input type="number" size="1" maxlength="6" 
+            v-model="newReaderPageInput"
+            @keydown.enter="setNewReader()"
+            :class="{
+              invalid: !isValidPageSet, 
+              empty: !newReaderPageInput || !newReaderPageInput.length, 
+              changed: newReaderPageChanged
+            }" >
+          <span :class="{ urgent: newReaderPageChanged }">
```
Comment (Bambosh) [inline src/components/SystemPages/NewReaderControls.vue:39]: This doesn't trigger for the Activate button, not sure why

File: src/components/SystemPages/NewReaderControls.vue [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+        <div style="height: 40px;" v-if="$isNewReader">
+            <br />
+          <div class="bigButtonRow">
+            <button @click="clearNewReader()">Switch off new reader mode</button>
+          </div>
+        </div>
+        <div style="height: 40px;" v-else>
+          <p class="hint">Or activate a preset:</p>
+          <div class="bigButtonRow">
+            <button @click="setupProblemSleuth()">Start Problem Sleuth</button>
```
Comment (Bambosh) [inline src/components/SystemPages/NewReaderControls.vue:61]: Switching from MSPA numbers to Viz numbers while a non-homestuck page is selected freaks the component out.
image

File: src/components/SystemPages/NewReaderControls.vue [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+      <dl class="fastForwardSelection">
+        <dt>
+          <input type="radio" id="fast_forward=false" :value="false" 
+            v-model="myFastForward"/>
+          <label for="fast_forward=false">Replay</label>
+        </dt>
+        <dd><label for="fast_forward=false">Read as if you were reading it live.<br>
+          All pages will be presented how they were as of the time of your most recent page. (with some minor exceptions; see 
+          <!-- Don't link to CC in setup mode (no settings page yet!) -->
+          <template v-if="false && $localData.assetDir">
```
Comment (Bambosh) [inline src/components/SystemPages/NewReaderControls.vue:103]: Perma-false - Confirm this works as intended?

**Review by Bambosh — CHANGES_REQUESTED** [review]

File: src/components/SystemPages/NewReaderControls.vue (hunk 4 above)
Comment (GiovanH) [reply src/components/SystemPages/NewReaderControls.vue:103]: stray debug code, whoops

File: src/components/SystemPages/NewReaderControls.vue (hunk 1 above)
Comment (GiovanH) [reply src/components/SystemPages/NewReaderControls.vue:150]: Bug, fixed

File: src/components/SystemPages/NewReaderControls.vue (hunk 2 above)
Comment (GiovanH) [reply src/components/SystemPages/NewReaderControls.vue:39]: Bug, fixed

File: src/components/SystemPages/NewReaderControls.vue (hunk 3 above)
Comment (GiovanH) [reply src/components/SystemPages/NewReaderControls.vue:61]: Fixed

**Review by Bambosh — APPROVED** [review]

**Pull request merged**